`[[deprecated]]` usage disabled for C++11

mpusz opened the pull request:
`[[deprecated]]` attribute was introduced in C++14. This library claims to be compatible with C++11. This is why it was replaced with `__attribute__` usage.

mpusz committed:
`[[deprecated]]` usage disabled for C++11

`[[deprecated]]` attribute was introduced in C++14. This library claims to be compatible with C++11. This is why it was replaced with `__attribute__` usage.

potswa commented:
Thanks!

Unfortunately I can’t accept pull requests right now as I’m outside my country without my laptop. 

Hopefully in a month or two.

Sent from my iPhone

> On 7 Apr 2020, at 02:08, Mateusz Pusz <notifications@github.com> wrote:
> 
> ﻿
> [[deprecated]] attribute was introduced in C++14. This library claims to be compatible with C++11. This is why it was replaced with __attribute__ usage.
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/potswa/cxx_function/pull/21
> 
> Commit Summary
> 
> `[[deprecated]]` usage disabled for C++11
> File Changes
> 
> M cxx_function.hpp (4)
> Patch Links:
> 
> https://github.com/potswa/cxx_function/pull/21.patch
> https://github.com/potswa/cxx_function/pull/21.diff
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.

potswa commented:
Can you access the CI results? It’s failing on MSVC, so the `attribute` usage needs to be limited to GCC.

mpusz committed:
Attribute `deprecated` disabled for C++11

mpusz commented:
> It’s failing on MSVC, so the attribute usage needs to be limited to GCC.

I hope I fixed it with the second commit?

--- edited ---

Nope, it did not. Made the third commit.

mpusz committed:
[[deprecated]] for C++11 disabled only for GCC

mpusz commented:
CI looks better now but still failing. Is it my issue?

potswa committed:
Take MSVC as the special case instead of GCC.

All vendors except MS implemented GCC's __attribute__ syntax,
and MSVC implemented the C++14 feature very early in C++11 mode.

potswa commented:
Thanks! Right, CI keeps failing because MSVC doesn't support the `scoped_allocator` testcase. I should probably just disable it…

I made a little adjustment so that your fix applies for all platforms except MSVC.
